code/modules/paperwork/clipboard.dm

```diff
@@ -94,42 +104,45 @@
 	else
 		dat += "<A href='?src=\ref[src];addpen=1'>Add Pen</A><BR><HR>"
 
-	for(var/i = 1 to length(papers))
+	dat += "<TABLE style='table-layout:fixed; white-space:nowrap;'>"
+	for(var/i = 1 to LAZYLEN(papers))
 		var/obj/item/P = papers[i]
-		dat += "<A href='?src=\ref[src];examine=\ref[P]'>[P.name]</A> - <DIV style='float:right;white-space: nowrap;'>"
+		dat += "<TR><TD style='width:45%; overflow:hidden; text-overflow:ellipsis;'><A href='?src=\ref[src];examine=\ref[P]'>[P.name]</A></TD>"
 		if(i == 1)
-			dat += "<A href='?src=\ref[src];write=\ref[P]'>Write</A> "
-		dat += "<A href='?src=\ref[src];remove=\ref[P]'>Remove</A> <A href='?src=\ref[src];rename=\ref[P]'>Rename</A></DIV><BR>"
+			dat += "<TD/><TD><A href='?src=\ref[src];write=\ref[P]'>Write</A></TD>"
+		else
+			dat += "<TD/><TD/>"
+		dat += "<TD><A href='?src=\ref[src];remove=\ref[P]'>Remove</A></TD><TD><A href='?src=\ref[src];rename=\ref[P]'>Rename</A></TD></TR>"
+	dat += "</TABLE>"
 
+	user.set_machine(src)
```

Comment:
I need to do it manually? Take a look at the code behind it? Its nothing new, look at the code for ``/obj/item/paicard`` for instance. 
If you wanna fix it and make a better system for keeping track of open non-nano guis then be my guest? But this isn't in the scope.

Reply:
Also take a look at ``/obj/proc/updateUsrDialog()`` and ``/obj/proc/updateDialog()``.
That's how browser windows that aren't nano handle state updates without just calling attack_self willy-nilly.

Reply:
UIs that do this do not manually set machine for the most part (yes, I know there are random exceptions). Machinery, the level at which `user.machine` is somewhat standardized, handle setting machine and searching for users using `CouldUseTopic`, the on-close listener, etc. In particular they are somewhat accurate at _clearing_ this var, which is relevant for GC. If you really need this functionality, you may wish to move that handling up a level or abstracting it to a flag, extension, etc. I don't like lots of ad hoc setting of it because there are in fact some best practices to follow here.

That said, yes, obviously the UI update system for non-nano UIs is pretty shit. Another alternative besides keeping weakrefs to users as I mentioned above is keeping refs to browser datums.

Reply:
Its not like I didn't try other alternatives first. The only problem I got with /datum/browser is that I'd need to hold refs per user inside the object.  And then handling the cleanup of closed windows would get really annoying, and I'd probably need to use onclose anyways. It seems to have been thought out for things that only one person may look at at a time.

Weakrefs are really the way to go imho. Its just I don't think piling up another custom handling over the current generic one is going to be worth it. At least using the old system, its way simpler to refactor the thing later on when someone makes a proper user handling system for browser window guis?